src/lib/sql-pool.ts

```diff
@@ -0,0 +1,103 @@
+import { createPool, Factory, Pool } from 'generic-pool';
+import { Config, ExasolDriver, websocketFactory } from './sql-client';
+import { ILogger, LogLevel, Logger } from './logger/logger';
+import { QueryResult } from './query-result';
+import { Attributes } from './commands';
+import { CetCancelFunction } from './sql-client.interface';
+import { SQLQueriesResponse, SQLResponse } from './types';
+//https://www.npmjs.com/package/generic-pool
+//https://www.npmjs.com/package/@types/generic-pool
+export interface PoolConfig {
+  min: number;
+  max: number;
+}
+export function getPool(websocketFactory: websocketFactory, config: Partial<Config> & Partial<PoolConfig>, logger: ILogger) {
+  // export interface Factory<T> {
+  //   create(): Promise<T>;
+  //   destroy(client: T): Promise<void>;
+  //   validate?(client: T): Promise<boolean>;
+  // }
+  async function createClient() {
+    const exasolClient: ExasolDriver = new ExasolDriver(websocketFactory, config, logger);
+    await exasolClient.connect();
+    return Promise.resolve(exasolClient);
```

Comment:
```suggestion
    return await exasolClient.connect();
```
Should do the same, not sure ;)

Reply:
Yep, it does, I don't think it's as readable but it does. I'll change it.